data/summary/types/cgm.go

```diff
@@ -231,6 +246,26 @@ func (s *CGMStats) Update(ctx context.Context, cursor fetcher.DeviceDataCursor)
 	return nil
 }
 
+func (B *CGMBucketData) CalculateVariance(value float64, duration float64) float64 {
+	var mean float64 = 0
+	if B.TotalMinutes > 0 {
+		mean = B.TotalGlucose / float64(B.TotalMinutes)
+	}
+
+	weight := float64(B.TotalMinutes) + duration
```

Comment:
This appears to be using the "weighted incremental algorithm" as detailed on Wikipedia at https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Weighted_incremental_algorithm

If so, consider calling that out in a comment. It helps reviewers double check that you've nailed the algorithm if we know what you were aiming for.

Reply:
Anything on the web can change at any time (think about archive.org's recentl legal cases and what impact they might have). The point is just to try and be helpful. It's much easier to verify this code does what you think it does when you know exactly what it's trying to do. :)

IMO linking to wikipedia is fine for this. Wikipedia has a history function, and things like this aren't too likely to change. It's not a hotly contested area of current research.

Wikipedia references this article: https://dl.acm.org/doi/10.1145/359146.359153 you could link that instead/too. It's from CACM which recently became free to anyone.

Reply:
Linked to the relevant wikipedia section in a comment.